.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,214 @@
+          SYMFONY_PATCH_TYPE_DECLARATIONS=force=1 php .github/patch-types.php # ensure the script is idempotent
+          echo PHPUNIT="$PHPUNIT,legacy" >> $GITHUB_ENV
+
+      - name: Run tests
+        run: |
+          _run_tests() {
+            local ok=0
+            local title="$1 $FLIP"
+            local start=$(date -u +%s)
+            OUTPUT=$(bash -xc "$2" 2>&1) || ok=1
+            local end=$(date -u +%s)
+
+            if [[ $ok -ne 0 ]]; then
+              printf "\n%-70s%10s\n" $title $(($end-$start))s
+              echo "$OUTPUT"
+              echo -e "\n\\e[41mKO\\e[0m $title\\n"
+            else
+              printf "::group::%-68s%10s\n" $title $(($end-$start))s
+              echo "$OUTPUT"
+              echo -e "\n\\e[32mOK\\e[0m $title\\n\\n::endgroup::"
+            fi
+
+            [[ "${{ matrix.mode }}" = experimental ]] || (exit $ok)
+          }
+          export -f _run_tests
+
+          if [[ ! "${{ matrix.mode }}" = *-deps ]]; then
+            echo "$COMPONENTS" | xargs -n1 | parallel -j +3 "_run_tests {} '$PHPUNIT {}'"
+
+            exit 0
+          fi
+
+          (cd src/Symfony/Component/HttpFoundation; cp composer.json composer.bak; composer require --dev --no-update mongodb/mongodb)
+
+          if [[ "${{ matrix.mode }}" = low-deps ]]; then
+            echo "$COMPONENTS" | xargs -n1 | parallel -j +3 "_run_tests {} 'cd {} && $COMPOSER_UP --prefer-lowest --prefer-stable && $PHPUNIT'"
+
+            exit 0
+          fi
+
+          # matrix.mode = high-deps
+          echo "$COMPONENTS" | xargs -n1 | parallel -j +3 "_run_tests {} 'cd {} && $COMPOSER_UP && $PHPUNIT$LEGACY'" || X=1
+
+          (cd src/Symfony/Component/HttpFoundation; mv composer.bak composer.json)
+          COMPONENTS=$(git diff --name-only src/ | grep composer.json || true)
+
+          if [[ $COMPONENTS && $SYMFONY_VERSION = *.4 ]]; then
+              export FLIP='🙃'
+              SYMFONY_VERSION=$(echo $SYMFONY_VERSION | awk '{print $1 - 1}')
+              echo -e "\\n\\e[33;1mChecking out Symfony $SYMFONY_VERSION and running tests with patched components as deps\\e[0m"
+              export COMPOSER_ROOT_VERSION=$SYMFONY_VERSION.x-dev
+              export SYMFONY_REQUIRE=">=$SYMFONY_VERSION"
+              git fetch --depth=2 origin $SYMFONY_VERSION
+              git checkout -m FETCH_HEAD
+              COMPONENTS=$(echo "$COMPONENTS" | xargs dirname | xargs -n1 -I{} bash -c "[ -e '{}/phpunit.xml.dist' ] && echo '{}'" | sort || true)
+              (cd src/Symfony/Component/HttpFoundation; composer require --dev --no-update mongodb/mongodb)
+              [[ ! $COMPONENTS ]] || echo "$COMPONENTS" | parallel -j +3 "_run_tests {} 'cd {} && rm composer.lock vendor/ -Rf && $COMPOSER_UP && $PHPUNIT$LEGACY'" || X=1
+          fi
+
+          [[ ! $X ]] || (exit 1)
+
+      - name: Run tests with SIGCHLD enabled PHP
+        if: "${{ matrix.php == '7.1' }}"
```

Comment:
@nicolas-grekas You have removed PHP 7.1 from the matrix. Because of that, this step is not executed at the moment.

Reply:
Right! Fixed in bb59651e48

Reply:
Thanks!